HW20/DeckGenerator.h

```diff
@@ -0,0 +1,16 @@
+#pragma once
+#include <vector>
+#include <cstdlib>
+#include <stdexcept>
+
+template <typename T>
+T generateRandomElement(const std::vector<T>& database) {
+    if (database.empty()) 
+    {
+        throw std::out_of_range("Database is empty");
+    }
+    unsigned int index = std::rand() % database.size();
+    return database[index];
+}
```

Comment:
They'll be duplicated and they shouldn't. Only when the deck is fully empty it should be reshuffled and started all over again